docs/synapse/userguides/storm_ref_filter.rstorm

```diff
@@ -0,0 +1,961 @@
+
+- ``+`` specifies an **inclusion** filter. The filter downselects the working set to **only** those nodes that match the specified criteria.
+- ``-`` specifies an **exclusion** filter. The filter downselects the working set to all nodes **except** those that match the specified criteria.
+
+The types of filter operations within Storm are highly flexible and consist of the following:
+
+- `Simple Filters`_
+- `Filters Using Standard Comparison Operators`_
+- `Filters Using Extended Comparison Operators`_
+- `Compound Filters`_
+- `Subquery Filters`_
+
+In most cases, the criteria and available comparators for lift operations (:ref:`storm-ref-lift`) are also available for filter operations.
+
+.. NOTE::
+   When filtering based on a secondary property (*<prop>*) or secondary property value (*<prop> = <pval>*), the property can be specified using either the relative property name (``:baz``) or the full form and property name (``foo:bar:baz``). Using the relative property name allows for simplified syntax and more efficient data entry within the cmdr CLI. Full property names may be required in cases where multiple nodes in the inbound node set have the same secondary property (e.g., ``inet:dns:a:ipv4`` and ``inet:url:ipv4``) and you only wish to filter based on the property / property value of one of the forms.
+   Where appropriate in the examples below, both syntaxes (full and relative) are provided for completeness / clarity. See the :ref:`data-props` section of :ref:`data-model-terms` for additional discussion of properties.
+
+See :ref:`storm-ref-syntax` for an explanation of the syntax format used below.
+
+See :ref:`storm-ref-type-specific` for details on special syntax or handling for specific data types.
+
+Simple Filters
+--------------
+
+"Simple" filters refers to the most "basic" filter operations: that is, operations to include ( ``+`` ) or exclude ( ``-`` ) a subset of nodes based on:
+
+- The presence of a specific primary or secondary property in the working set.
+- The presence of a specific primary property value or secondary property value in the working set.
+
+**Note:** :ref:`filter-tag` is addressed below.
+
+The only difference between "simple" filters and "filters using comparison operators" is that we have defined simple filters as those that use the equals ( ``=`` ) comparator, which is the easiest comparator to use to explain basic filtering concepts.
+
+**Syntax:**
+
+*<query>* **+** | **-** *<form>* [ **=** *<valu>* ] 
+
+*<query>* **+** | **-** [ *<form>* **=** *<valu>* ] **:** *<prop>* [ **=** *<pval>* ]  
+
+**Examples:**
+
+*Filter by Form (<form>):*
+
+- Downselect to include only domains:
+
+
+.. storm-pre:: [inet:fqdn=woot.com inet:fqdn=vertex.link  inet:fqdn=google.com inet:ipv4=127.0.0.1 ] +inet:fqdn
+.. storm-opts:: {"show": ["err", "fini", "init", "warn"]}
```

Comment:
We don't need to be sprinkling this directive around unless we're using it.

Reply:
removed